be1-go/channel/coin/verification.go

```diff
@@ -0,0 +1,77 @@
+package coin
+
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"golang.org/x/xerrors"
+	"popstellar/message/messagedata"
+	"strconv"
+)
+
+func (c *Channel) verifyMessageTransactionPost(transactionPost messagedata.TransactionPost) error {
+	c.log.Info().Msgf("verifying transaction#post message transaction_id validity %s",
+		transactionPost.TransactionId)
+
+	locktime := strconv.Itoa(transactionPost.Transaction.Locktime)
+	locktimeLen := strconv.Itoa(len(locktime))
+
+	version := strconv.Itoa(transactionPost.Transaction.Version)
+	versionLen := strconv.Itoa(len(version))
+
+	var transactionIdBefHash = locktimeLen + locktime
+
+	for _, inp := range transactionPost.Transaction.Inputs {
+		pubKey := inp.Script.PubKey
+		pubKeyLen := strconv.Itoa(len(pubKey))
+		transactionIdBefHash += pubKeyLen + pubKey
+
+		sig := inp.Script.Sig
+		sigLen := strconv.Itoa(len(sig))
+		transactionIdBefHash += sigLen + sig
+
+		typee := inp.Script.Type
+		typeeLen := strconv.Itoa(len(typee))
+		transactionIdBefHash += typeeLen + typee
+
+		hash := inp.Hash
+		hashLen := strconv.Itoa(len(hash))
+		transactionIdBefHash += hashLen + hash
+
+		index := strconv.Itoa(inp.Index)
+		indexLen := strconv.Itoa(len(index))
+		transactionIdBefHash += indexLen + index
+	}
+
+	for _, out := range transactionPost.Transaction.Outputs {
+		pubKey := out.Script.PubKeyHash
+		pubKeyLen := strconv.Itoa(len(pubKey))
+		transactionIdBefHash += pubKeyLen + pubKey
+
+		typee := out.Script.Type
+		typeeLen := strconv.Itoa(len(typee))
+		transactionIdBefHash += typeeLen + typee
+
+		value := strconv.Itoa(out.Value)
+		valueLen := strconv.Itoa(len(value))
+		transactionIdBefHash += valueLen + value
+	}
+
+	transactionIdBefHash += versionLen + version
+	computedTransactionId := Hash(transactionIdBefHash)
+
+	if transactionPost.TransactionId != computedTransactionId {
+		return xerrors.Errorf("Transaction Id is not valid, value=%s, computed=%s", transactionPost.TransactionId, computedTransactionId)
+	}
+
+	return nil
+}
+
+func Hash(strs ...string) string {
+	h := sha256.New()
+	for _, s := range strs {
+		//h.Write([]byte(fmt.Sprintf("%d", len(s))))
+		h.Write([]byte(s))
+	}
+
+	return base64.URLEncoding.EncodeToString(h.Sum(nil))
+}
```

Comment:
The hashing scheme is designed so that we can reuse `messageData.Hash` as-is. Doing it this way is less maintainable, so I would recommend refactoring to use it instead of calling `strconv.Itoa` all over the place.

Reply:
Yes I understand your comment, the only thing that I don't know how to do is how to deal with the array Input[].
Indeed as we don't know the length of the Input array it is complicated to deal with it. (Unless I've missed something)
Don't hesitate to tell me how you would have done. 😀

Reply:
I'm confused as to why the length of that array is an issue. It's not included in the hash. We just process every one of its elements one after the other...

Reply:
See #992 you can pull it

Reply:
Ohh okay, it's my fault, I didn't read the go doc well enough. I thought these array []string were fixed size (as in c).
Thanks for your time and your help!